### DISCUSSION TITLE
Could you use supabase from a CLI

### DISCUSSION THREAD

**Discussion by jaredcosulich** [supabase/supabase#5803] [Questions]
Would it be feasible to use Supabase from a CLI instead of a webapp? How would you authenticate?

**Comment by GaryAustin1** [comment]
You might clarify what you are looking to do from the CLI.

Supabase provides a connection to the Postgres data base with a user and password.  You can run any postgres data base tool including PSQL terminal with this connection.

**Reply by jaredcosulich** [reply]
Yes, sorry, we would be interested in taking advantage of the row-level security on the database as well as storage. The CLI would be used by our users so we would want to treat it as if it were the "clientside" of a website. Does that make sense?

**Reply by GaryAustin1** [reply]
I've not seen a direct example of a CLI to the Supabase API.  But it seems it would be just issuing http API calls to Auth to signup/signin users and get tokens.  Then making http API calls to the database (which uses postgrest) and storage which would go thru RLS.  Besides handling token refreshes, supabase.js (and its component modules) just convert things to http API calls.  

I would think any of the discussions involving servers (like next.js and SSR on other platforms) would address the issues you would face.  
From what I have seen managing the session tokens and refreshing is the hardest part, but alot of those are also dealing with browser client code in the mix too.

I'm not of much more help, I recommend you search around a bit on "server" "api" "supabase"  and if no one else drops in here maybe re-ask with more detail.

https://supabase.com/docs/guides/api
Towards to bottom of this is auth api info:  https://supabase.com/docs/gotrue/server/about
https://PostgREST.org also has  info on its http API.

**Reply by jaredcosulich** [reply]
Thank you for the thoughts. I'll dig in more given the possibility it might work.
